common/src/include/ebbrt/MulticoreEbb.h

```diff
@@ -11,10 +11,42 @@
 
 #include <ebbrt/Cpu.h>
 #include <ebbrt/EbbAllocator.h>
+#include <ebbrt/EbbId.h>
 #include <ebbrt/EbbRef.h>
 #include <ebbrt/LocalIdMap.h>
 
 namespace ebbrt {
+
+template <class T> class MulticoreEbbRoot {
+ public:
+  struct RepMapRef {
+    LocalIdMap::ConstAccessor accessor;
+    const boost::container::flat_map<size_t, T*>* ref;
+  };
+  MulticoreEbbRoot() = default;
+  explicit MulticoreEbbRoot(EbbId id) : id_{id} {};
+
+  void SetEbbId(EbbId id) { id_ = id; }
+  RepMapRef* GetReps() const {
+    if (!id_)
+      throw std::runtime_error("No EbbRT specified in MulticoreEbbRoot");
+    auto rep = new RepMapRef();
```

Comment:
Does this have to be heap allocated? Why is it returned as a pointer? At the very least it should be unique_ptr if not by value.